src/o365/spo/commands/userprofile/userprofile-set.ts

```diff
@@ -0,0 +1,134 @@
+import { ContextInfo } from '../../spo';
+import request from '../../../../request';
+import commands from '../../commands';
+import GlobalOptions from '../../../../GlobalOptions';
+import {
+  CommandOption,
+  CommandValidate
+} from '../../../../Command';
+import SpoCommand from '../../../base/SpoCommand';
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  userName: string;
+  propertyName: string;
+  propertyValue: string;
+}
+
+class SpoUserprofileSetCommand extends SpoCommand {
+  public get name(): string {
+    return commands.USERPROFILE_SET;
+  }
+
+  public get description(): string {
+    return 'Sets user profile property for a SharePoint user';
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: (err?: any) => void): void {
+    let spoUrl: string = '';
+
+    this
+      .getSpoUrl(cmd, this.debug)
+      .then((_spoUrl: string): Promise<ContextInfo> => {
+        spoUrl = _spoUrl;
+
+        return this.getRequestDigest(spoUrl);
+      })
+      .then((res: ContextInfo): Promise<string> => {
+        const propertyValue: string | string[] = args.options.propertyValue.split(',').map(o => o.trim());
+        let propertyType: string = 'SetSingleValueProfileProperty';
+        const body: any = {
+          'accountName': `i:0#.f|membership|${args.options.userName}`,
+          'propertyName': args.options.propertyName,
+        };
+
+        if (propertyValue.length > 1) {
+          propertyType = 'SetMultiValuedProfileProperty';
+          body.propertyValues = [...propertyValue];
+        } else {
+          body.propertyValue = propertyValue[0];
+        }
+
+        const requestOptions: any = {
+          url: `${spoUrl}/_api/SP.UserProfiles.PeopleManager/${propertyType}`,
+          headers: {
+            'Accept': 'application/json;odata=nometadata',
+            'Content-type': 'application/json;odata=verbose',
+            'X-RequestDigest': res.FormDigestValue
+          },
+          body: JSON.stringify(body)
```

Comment:
Rather than serializing the request body ourselves, we should pass the original object and set the `json: true` property on the request to have it handle the serialization automatically for us.